src/baal/modelwrapper.py

```diff
@@ -335,17 +337,39 @@ def predict_on_batch(self, data, iterations=1, cuda=False):
         Returns:
             Tensor, the loss computed from the criterion.
                     shape = {batch_size, nclass, n_iteration}
+
+        Raises:
+            raises RuntimeError if CUDA rans out of memory during data replication.
         """
         with torch.no_grad():
             if cuda:
                 data = data.cuda()
-            input_shape = data.size()
-            batch_size = input_shape[0]
-            data = torch.stack([data] * iterations)
-            data = data.view(batch_size * iterations, *input_shape[1:])
-            out = self.model(data)
-            out = map_on_tensor(lambda o: o.view([iterations, batch_size, *o.size()[1:]]), out)
-            out = map_on_tensor(lambda o: o.permute(1, 2, *range(3, o.ndimension()), 0), out)
+            if self.replicate_in_memory:
+                input_shape = data.size()
+                batch_size = input_shape[0]
+                try:
+                    data = torch.stack([data] * iterations)
+                except Exception as e:
```

Comment:
Shouldn't we restrict this to RuntimeError? Just catching every exception doesn't seem like the safest thing to me, both here but especially in line 359 since there can be all kinds of errors when passing data through the model.

Catching runtime error and verifying that it is an out of memory error seems safer.

Reply:
Yes, absolutely, good catch. My intention was to actually catch a RuntimeError there but somehow I missed the initial 'catch-all' change.